pkg/utils/experimentHelpers.go

```diff
@@ -109,39 +111,42 @@ func (expDetails *ExperimentDetails) SetEnvFromEngine(engineName string, clients
 }
 
 // SetLabels sets the Experiment Labels, in Experiment Structure
-func (expDetails *ExperimentDetails) SetLabels(clients ClientSets) {
+func (expDetails *ExperimentDetails) SetLabels(engine *EngineDetails, clients ClientSets) error {
 	expirementSpec, err := clients.LitmusClient.LitmuschaosV1alpha1().ChaosExperiments(expDetails.Namespace).Get(expDetails.Name, metav1.GetOptions{})
 	if err != nil {
-		klog.V(0).Infoln(err)
+		return errors.Wrapf(err, "Unable to get ChaosExperiment instance in namespace: %v", expDetails.Namespace)
 	}
 	expDetails.ExpLabels = expirementSpec.Spec.Definition.Labels
-
+	expDetails.ExpLabels["chaosUID"] = engine.UID
```

Comment:
This  value is engineUID and setLabels is called in setValueFromChaosExperiment, why are we taking the value here?

Reply:
Please move it to setValueFromChaosEngine

Reply:
Because this function is written to set the labels of chaosExperiments. We can't create multiple functions for the same thing (only to set labels)